synapse/handlers/profile.py

```diff
@@ -283,6 +288,47 @@ def _update_join_states(self, requester, target_user):
                     room_id, str(e)
                 )
 
+    @defer.inlineCallbacks
+    def check_profile_query_allowed(self, target_user, requester=None):
+        """Checks whether a profile query is allowed. If the
+        'require_auth_for_profile_requests' config flag is set to True and a
+        'requester' is provided, the query is only allowed if the two users
+        share a room.
+
+        Args:
+            target_user (UserID): The owner of the queried profile.
+            requester (None|UserID): The user querying for the profile.
+
+        Raises:
+            SynapseError(403): The two users share no room, or ne user couldn't
+                be found to be in any room the server is in, and therefore the query
+                is denied.
+        """
+        # Implementation of MSC1301: don't allow looking up profiles if the
+        # requester isn't in the same room as the target. We expect requester to
+        # be None when this function is called outside of a profile query, e.g.
+        # when building a membership event. In this case, we must allow the
+        # lookup.
+        if self.hs.config.require_auth_for_profile_requests and requester:
+            try:
+                requester_rooms = yield self.store.get_rooms_for_user(
+                    requester.to_string()
+                )
+                target_user_rooms = yield self.store.get_rooms_for_user(
+                    target_user.to_string(),
+                )
+
+                # Check if the length of the intersection between the room lists
+                # for both users is 0.
+                if not len(requester_rooms.intersection(target_user_rooms)):
```

Comment:
`len` is redundant: an empty set is falsy. In any case I think you want `if requester_rooms.isdisjoint(target_user_rooms)` to avoid making a redundant set.

Reply:
Thanks for the tip!